### DISCUSSION TITLE
Looking for clarification on the difference between compare-options and resource.exclusions

### DISCUSSION THREAD

**Discussion by tho93** [argoproj/argo-cd#13338] [Q&A]
Hello all,

I'm hoping to get some clarity on the differences between [Compare Options](https://argo-cd.readthedocs.io/en/stable/user-guide/compare-options/#ignoring-resources-that-are-extraneous) and [Resource Exclusions](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#resource-exclusioninclusion). Namely, it seems like resource exclusions will allow ArgoCD to exclude certain resources from both discovery AND sync, whereas compare-options only applies when ArgoCD is showing sync status. What is the difference between discovery and sync?

The motivation behind this question is that I have been noticing intermittent issues with the following error message:
```
Failed to cache app resources: error getting resource tree: error synchronizing cache state : failed to sync cluster https://<server_ip>:443: failed to load initial state of resource PersistentVolumeClaim: The provided continue parameter is too old to display a consistent list result. You can start a new list without the continue parameter, or use the continue token in this response to retrieve the remainder of the results. Continuing with the provided token results in an inconsistent list - objects that were created, modified, or deleted between the time the first chunk was returned and now may show up in the list.
```

My cluster has a high number of PVCs provisioned by the [EBS CSI](https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html). I would like for Argo to ignore a subset of PersistentVolumeClaims (the claims generated by the ebs csi). From the documentation, it appears as though this can be achieved by setting the `argocd.argoproj.io/compare-options: IgnoreExtraneous` annotation on the PVCs I want ignored.

From searching through the GitHub issues page, I found a very similar issue [here](https://github.com/argoproj/argo-cd/issues/6113). The advice given in this post recommends setting `resource.exclusions`. However, I'm not sure if I want the PVCs to be excluded from discover, sync, or both (or some other mechanisms I'm not aware of).

Thanks for all the help!

**Comment by andrii-korotkov-verkada** [comment]
Compare options apply to determine if the application is in sync or not. 
`argocd.argoproj.io/compare-options: IgnoreExtraneous` would let your app to appear in sync even if there are some undeleted resources with this annotation. 

Resource exclusions go way lower level and configure what Kubernetes resources are being watched by ArgoCD. For example, if the object is updated in k8s but excluded, it won't trigger an app reconciliation and I think won't even show up in UI.

Sounds like your concern's reason is due to slow processing times of watched resources when setting up a watch (by default happens every 10 minutes per resource kind). Can you give an estimate of the scale, please? I.e. how many PVCs do you have in your cluster and how many updates are there? I now have a similar question regarding Node resources which are managed by Karpenter and they don't even show up in the app UI. Overall, it seems like exclusions is the best bet for you and it may be okay for this kind of resource, but you'd need to test.
